src/noc/containers/noc-check-and-submit/noc-check-and-submit.component.ts

```diff
@@ -56,6 +56,11 @@ export class NocCheckAndSubmitComponent implements OnInit, OnDestroy {
                 return questions.find(ques => ques.question_id === answer.question_id).question_text;
               }
             )),
+            question_type: this.questions$.pipe(map(
+              questions => {
+                return questions.find(ques => ques.question_id === answer.question_id).answer_field_type.type;
```

Comment:
Are you _guaranteed_ to find a question in this statement? If not, it will throw an exception when you try to access `.answer_field_type.type` on `undefined`.

Reply:
yes it will not be undefined